java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/ConnectionMutualTlsTest.java

```diff
@@ -0,0 +1,436 @@
+   * just a connection url and using 0 and 1 as ssl values.
+   *
+   * @throws Exception on error.
+   */
+  @Test
+  public void testTLSConnectionPropertyTrueIntegerCorrectCastUrlWithDriverManager()
+      throws Exception {
+    final Driver driver = new ArrowFlightJdbcDriver();
+    DriverManager.registerDriver(driver);
+
+    final String jdbcUrl = String.format(
+            "jdbc:arrow-flight-sql://localhost:%s?user=%s&password=%s" +
+                    "&useEncryption=1&useSystemTrustStore=0&%s=%s&%s=%s&%s=%s",
+            FLIGHT_SERVER_TEST_RULE.getPort(),
+            userTest,
+            passTest,
+            ArrowFlightConnectionProperty.TLS_ROOT_CERTS.camelName(),
+            URLEncoder.encode(tlsRootCertsPath, "UTF-8"),
+            ArrowFlightConnectionProperty.CLIENT_CERTIFICATE.camelName(),
+            URLEncoder.encode(clientMTlsCertPath, "UTF-8"),
+            ArrowFlightConnectionProperty.CLIENT_KEY.camelName(),
+            URLEncoder.encode(clientMTlsKeyPath, "UTF-8"));
+
+    try (Connection connection = DriverManager.getConnection(jdbcUrl)) {
+      Assert.assertTrue(connection.isValid(0));
+    }
+  }
+
+  /**
+   * Check if an encrypted connection can be established successfully when connecting through the DriverManager using
+   * a connection url and properties with String K-V pairs and using 0 and 1 as ssl values.
+   *
+   * @throws Exception on error.
+   */
+  @Test
+  public void testTLSConnectionPropertyTrueIntegerCorrectCastUrlAndPropertiesUsingSetPropertyWithDriverManager()
+      throws Exception {
+    final Driver driver = new ArrowFlightJdbcDriver();
+    DriverManager.registerDriver(driver);
+
+    Properties properties = new Properties();
+
+    properties.setProperty(ArrowFlightConnectionProperty.USER.camelName(), userTest);
+    properties.setProperty(ArrowFlightConnectionProperty.PASSWORD.camelName(), passTest);
+    properties.setProperty(ArrowFlightConnectionProperty.TLS_ROOT_CERTS.camelName(), tlsRootCertsPath);
+    properties.setProperty(ArrowFlightConnectionProperty.USE_ENCRYPTION.camelName(), "1");
+    properties.setProperty(ArrowFlightConnectionProperty.CLIENT_CERTIFICATE.camelName(), clientMTlsCertPath);
+    properties.setProperty(ArrowFlightConnectionProperty.CLIENT_KEY.camelName(), clientMTlsKeyPath);
+
+    final String jdbcUrl = String.format("jdbc:arrow-flight-sql://localhost:%s", FLIGHT_SERVER_TEST_RULE.getPort());
+
+    try (Connection connection = DriverManager.getConnection(jdbcUrl, properties)) {
+      Assert.assertTrue(connection.isValid(0));
+    }
+  }
+
+  /**
+   * Check if an encrypted connection can be established successfully when connecting through the DriverManager using
+   * a connection url and properties with Object K-V pairs and using 0 and 1 as ssl values.
+   *
+   * @throws Exception on error.
+   */
+  @Test
+  public void testTLSConnectionPropertyTrueIntegerCorrectCastUrlAndPropertiesUsingPutWithDriverManager()
```

Comment:
We could use `@ParameterizedTest` to reduce some code duplication in the test files.